pkgs/by-name/pdfding/package.nix

```diff
@@ -0,0 +1,194 @@
+    django-cleanup
+    django-htmx
+    gunicorn
+    markdown
+    minio
+    nh3
+    psycopg2-binary
+    pypdf
+    pypdfium2
+    python-magic
+    qrcode
+    rapidfuzz
+    ruamel-yaml
+    whitenoise
+    huey
+    supervisor # required but not used by the module, using systemd instead
+    pillow
+    oauthlib
+
+    # dependecies required for django collectstatic
+    requests
+    pyjwt
+    cryptography
+  ];
+
+  frontend = callPackage ./frontend.nix { };
+
+  pythonPath = python.pkgs.makePythonPath pythonPackages;
+in
+
+python.pkgs.buildPythonApplication rec {
+  pname = "pdfding";
+  version = "1.4.1";
+  src = fetchFromGitHub {
+    owner = "mrmn2";
+    repo = "PdfDing";
+    tag = "v${version}";
+    hash = "sha256-8e80gMdB6U3977dIU7bIAAEguYmi+AWQgUgYPDLCYLI=";
+  };
+  pyproject = true;
+
+  patches = [
+    # remove in 1.4.2 (next version after 1.4.1)
+    # patch to add data_dir
+    # https://github.com/mrmn2/PdfDing/pull/202
+    (fetchpatch2 {
+      url = "https://github.com/mrmn2/PdfDing/commit/387ca2079f74844203e2e91fac00e0d0e0e5fdb9.patch?full_index=1";
+      hash = "sha256-VGjyIAVi+qd2WZ8FVKKC2ijLinoflO7RmPwIW1/oGcY=";
+    })
+    # pyproject.toml still has 0.1.1 very old version
+    # follow https://github.com/mrmn2/PdfDing/pull/203
+    ./add-version.patch
+  ];
+
+  dependencies = pythonPackages;
+
+  build-system = with python.pkgs; [ poetry-core ];
+
+  nativeBuildInputs = [
+    makeWrapper
+  ];
+
+  preBuild = ''
+    # remove originals, copy from frontend
+    rm -rf pdfding/static
+    ln -s ${passthru.frontend}/pdfding/static pdfding/static
+
+    # not generating staticfiles.json if it exists
+    mv pdfding/core/settings/dev.py pdfding/core/settings/dev.py.bak
+
+    ${python.pythonOnBuildForHost.interpreter} pdfding/manage.py collectstatic
+
+    # dev.py is required so that test will run properly, restore it
+    mv pdfding/core/settings/dev.py.bak pdfding/core/settings/dev.py
+
+    # not needed, now we have staticfiles directory
+    rm -rf pdfding/static
+
+    # remove django md5 hash from filenames of pdfjs as it will mess up the relative imports because of the whitenoise setup
+    sh -x \
+      && export PDFJS_PATH="pdfding/staticfiles/pdfjs" \
+      && for file_name in $(find $PDFJS_PATH -type f -not -path "$PDFJS_PATH/web/images/*");  \
+        do \
+          if [[ $file_name =~ "LICENSE" ]]; then \
+            new=$(echo "$file_name" | sed -E "s/LICENSE\.[a-zA-Z0-9]{12}/LICENSE/"); \
+          else \
+            new=$(echo "$file_name" | sed -E "s/\.[a-zA-Z0-9]{12}\./\./"); \
+          fi; \
+          mv -- "$file_name" "$new"; \
+        done \
+      && echo 'Successfully removed hash from pdfjs files'
+
+    echo "VERSION = '${version}'" > pdfding/core/settings/version.py;
+  '';
+
+  postInstall = ''
+    mkdir -p $out/{bin,share}
+    pdfdingDir=$out/${python.sitePackages}/pdfding
+
+    # make an empty dir to supress the warning
+    mkdir -p $pdfdingDir/static
```

Comment:
I don't see a warning when this is missing.

Reply:
maybe it was happening in 1.3.3 when I first started packaging it, I can take a look at it, but its fine to leave it as is, I think.